### PR TITLE
Re-added vending_ammo to vending machines [Fixes #17081][Comments Requested]

### LINKED ISSUE #17081: Vending Machines in Gun Stores don't have any Ammunition/Magazines in them.
As the title says. The Issue seemed to emerge around Experimental 4771, with one person reporting it. I was able to duplicate the issue with Version 4834 back then [9th of May] and assumed that the Person who originally found the Issue reported it to github.
This does not seem to be the case however, as I just found out: The Vending machines are always empty it seems, and I spawned several gun stores, just to check if it was coincidence. Doesn't seem to be the case.
I'm currently using Version 5001 ; Maybe someone deleted the spawns from the vending machines when you added magazines?

### PR DESCRIPTION
**Pull request by BorkBorkGoesTheCode**
I have also updated the vending machine spawn lists.

### CODE REVIEW

File: data/json/itemgroups/ammo.json [hunk 1]
```diff
@@ -237,6 +237,78 @@
     ]
   },
   {
+    "type": "item_group",
+    "id": "vending_ammo_pistol",
```
Comment (mugling) [inline data/json/itemgroups/ammo.json]: `ammo_vending_pistol`

File: data/json/itemgroups/ammo.json [hunk 2]
```diff
@@ -237,6 +237,78 @@
...
+    "items": [
```
Comment (mugling) [inline data/json/itemgroups/ammo.json]: alphabetical ordering

File: data/json/itemgroups/ammo.json [hunk 3]
```diff
@@ -237,6 +237,78 @@
...
+      [ "22_fmj", 9 ],
```
Comment (mugling) [inline data/json/itemgroups/ammo.json]: round numbers (10, 20, 25) not arbitrary

File: data/json/itemgroups/ammo.json [hunk 4]
```diff
@@ -237,6 +237,78 @@
...
+      [ "22_lr", 27 ],
+      [ "22_cb", 24 ],
```
Comment (mugling) [inline data/json/itemgroups/ammo.json]: I'm not sure I like the requirement to provide a duplicate definition for some (majority) ammo types. You might best having vending machines dispense `ammo_common`. Having more disparate groups isn't of much benefit for a much higher maintenance cost.

**Comment by mugling** [comment]
Downvote is for the approach, not the concept of ammo vending machines. I don't think a completely separate list of item spawns is a good plan and is less likely to be updated than just using the `ammo_*_common` groups.

**Comment by Rivet-the-Zombie** [comment]
Yeah, I'd prefer to see this done using extant ammunition groups if possible.

**Comment by BorkBorkGoesTheCode** [comment]
Is it necessary to have multiple instances of an ammo group with different spawn points, or can I specify multiple spawn points for the same ammo group?

**Comment by BorkBorkGoesTheCode** [comment]
I have added ammo_common, ammo_rare, and ammo_milspec to the gunstore instead of vending_etc..

**Pull request closed without merging**